toolkit/tools/graphpkgfetcher/graphpkgfetcher.go

```diff
@@ -185,13 +313,206 @@ func resolveGraphNodes(dependencyGraph *pkggraph.PkgGraph, inputSummaryFile, out
+	timestamp.StartEvent("downloading delta nodes", nil)
+	defer timestamp.StopEvent(nil)
+
+	// First scan the copy of the graph we tried to optimize for all the SRPMs we need to build. We will use this list to
+	// match against all the nodes in the full graph.
+	srpmPaths := make(map[string]bool)
+	for _, n := range dependencyGraphDeltaCopy.AllBuildNodes() {
+		srpmPaths[n.SrpmPath] = true
+	}
+
+	// We don't want to download implicit nodes since they will be included in another node with the same SRPM. Keep a list
+	// of them so we can fix them up later.
+	skippedNodes := []*pkggraph.PkgNode{}
+	// We will need to keep track of the original path to delta path mapping so we can fix up the implicit nodes later.
+	originalPathToDeltaPathMap := make(map[string]string)
+
+	// For each build node, try to update it to a delta node with a downloaded RPM backing it.
+	logger.Log.Debugf("Resolving build nodes")
+	for _, n := range realDependencyGraph.AllBuildNodes() {
+		// Implicit nodes cause us troubles since we don't know exactly which RPMs they will build (so the cache fetcher
+		// will pull all of the possible matches). Since we are already matching against SRPMs, we can safely skip these
+		// nodes since they will be included in another node with the same SRPM.
+		if n.Implicit {
+			logger.Log.Debugf("Skipping implicit delta build node %s", n)
+			skippedNodes = append(skippedNodes, n)
+			continue
+		}
+
+		logger.Log.Debugf("Resolving build node %s", n)
+		if n.State == pkggraph.StateBuild {
+			foundMatch, err := downloadSingleDeltaRPM(realDependencyGraph, n, cloner, *outDir, &originalPathToDeltaPathMap)
+			if err != nil {
+				return fmt.Errorf("failed to download delta RPM for build node %s: %w", n, err)
+			}
+			if !foundMatch {
+				// Throw any nodes we fail to resolve the fist time into a list so we can try to resolve them again later.
+				// This will help with cases were a new sub-package is added to a build node, but we won't be able to download
+				// the delta RPM since it doesn't exist yet.
+				skippedNodes = append(skippedNodes, n)
+			}
+		}
+	}
 
-	if strings.TrimSpace(outputSummaryFile) != "" {
-		err = repoutils.SaveClonedRepoContents(cloner, outputSummaryFile)
+	// Fix up the implicit nodes to point to the correct delta RPMs that we parsed earlier.
+	for _, n := range skippedNodes {
+		logger.Log.Debugf("Fixing up skipped node %s", n)
+		err = fixupDeltaImplicitNode(realDependencyGraph, n, &originalPathToDeltaPathMap)
 		if err != nil {
-			logger.Log.Errorf("Failed to save cloned repo contents.")
-			return
+			return fmt.Errorf("failed to fixup skipped node %s: %w", n, err)
+		}
+	}
+
+	logger.Log.Info("Configuring additional delta RPMs as a local repository")
+	err = cloner.ConvertDownloadedPackagesIntoRepo()
+	if err != nil {
+		logger.Log.Errorf("Failed to convert downloaded RPMs into a repo. Error: %s", err)
+		return
+	}
```

Comment:
Don't we already run `ConvertDownloadedPackagesIntoRepo` after this function is done anyway?

Reply:
Good catch, this was an artifact from merging in the other PR's changes.